### PR TITLE
Sup 18 adjust error handling

### PR DESCRIPTION
**Pull request by tehnorm**
Adjust the error handling to throw exceptions vs the error/data array.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .phpunit.result.cache
 /vendor
+/tests/fixtures
```
Comment (tehnorm) [inline .gitignore:3]: Did we mean to ignore the fixtures from PHP-VCR?

**Review by tehnorm — COMMENTED** [review]
Looks good - we will want to make sure to get the fixtures back into the directory so the tests run correctly.

File: .gitignore (hunk 1 above)
Comment (adolfo1971) [reply .gitignore:3]: It has already been put so that it is not ignored

File: src/Storage/StorageBucket.php [hunk 2]
```diff
@@ -117,9 +148,10 @@ public function deleteBucket($bucketId)
 	}
 
 	/**
-	 * Removes all objects inside a single bucket.
+	 *	Removes all objects inside a single bucket.
```
Comment (tehnorm) [inline src/Storage/StorageBucket.php:151]: Extra space here?

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+#### Handling Files
+
+### Connecting to the storage backend
+
+```php
+  include '../vendor/autoload.php';
+
+  use Supabase\Storage\StorageFile;
+
+  $authHeader = ['Authorization' => 'Bearer '.'<your_api_key>'];
```
Comment (tehnorm) [inline examples/README.md:109]: We may want to use string interpolation here:
```
$api_key = '12345';
$authHeader = ['Authorization' => "Bearer {$api_key}"];
```

This should apply to all places we are doing this in the repo.

File: .gitignore (hunk 1 above)
Comment (tehnorm) [inline .gitignore:3]: We need to remove this line, rerun PHP VCR and then commit the new fixtures so that we have them in the Repo.

**Review by tehnorm — COMMENTED** [review]
Solid work. We need to make a few changes to get this ready for merge.

**Pull request merged**